### PR TITLE
external endpoint timeout guard

### PR DESCRIPTION
**Pull request by S0naliThakur**
Linear: https://linear.app/shm/issue/PRI-41
Summary: Added middleware to enforce a timeout on external endpoints, preventing waits beyond a set time limit.

### CODE REVIEW

File: src/network/index.ts [hunk 1]
```diff
@@ -162,11 +166,33 @@ export class NetworkClass extends EventEmitter {
         next()
       }
 
+      const timeoutMiddleware = function (req, res, next) {
+        const timeout = config.network.requestTimeout * 1000 || 5000
```
Comment (jintukumardas) [inline src/network/index.ts:170]: We can remove the  `|| 5000` option.

**Review by jintukumardas — CHANGES_REQUESTED** [review]

File: src/network/index.ts (hunk 1 above)
Comment (S0naliThakur) [reply src/network/index.ts:170]: network in config is optional. 5000 is fallback

**Review by jintukumardas — DISMISSED** [review]

**Comment by kgmyatthu** [comment]
This can be done in much simpler way. Please use express's builtin support for this. Something like this is all you need.
Example: 

```javascript
const server = app.listen()
server.setTimeout(5000)
```
Current implementation in PR add unnecessary LOC(s). Please modify.

**Review by kgmyatthu — COMMENTED** [review]
Please check the closing statement

**Pull request closed without merging**